interview-questions.md

```diff
@@ -9,59 +9,95 @@ Then, research the question to expand on your answer. Even if you feel you have
+                and output a manipulation thereof.
 
-  Researched answer:
+  Researched answer: In Javascript the 'this' keyword refers to the object it belongs to. As a method, it refers to the owner object. As a function, it refers to
+                      the global. In an event, 'this' refers to the corresponding element that fired the event. In strict mode, 'this' is undefined.
 
 
 
 2. What is yarn? What file(s) are modified in a React application when you run yarn?
 
-  Your answer:
+  Your answer: yarn is a package manager for Javascript libraries. It installs and maintains packages based on a package.json file that describes the name of each
+                module and the desired version number of said module. If no version number is specified, it typically defaults to the latest stable release. Whenever
+                yarn installs a dependency, it will add the name and actual version number installed to the yarn.lock file. You can modify the package.json and run $ yarn
+                to install a new/updated plugin, or you can use $ yarn add ${plugin name here}. In either scenario, yarn will attempt to install the package and add the result
+                to the yarn.lock file.
 
-  Researched answer:
+  Researched answer: Yarn is a package manager that also fills the role of a project manage. It was developed by Facebook after failing to adequately refactor npm
+                    for their needs. Refactoring npm was slow and difficult, so Facebook decided to just do it themselves. It effectively replaces npm as a package manager.
+                    Yarn specifically checks first that all its dependencies are compatible, then looks into the global cache to see if it needs to download anything to
+                    bring the package up to date with the .json file. If it does, it fetches all the data for the package and places is in a cache, then finally moves
+                    all this information into the appropriate node_modules folder and links it together with the project.
 
 
 
 3. What are props in React?
 
-  Your answer:
+  Your answer: props is short for properties, at least in my head. They are values and methods that are passed from a parent class to a child class. More specifically,
+              props is an object that a child class has access to that contains all the information from the parent class, including all its helper methods and data.
 
-  Researched answer:
+  Researched answer: When React sees an element in JSX representing a user-defined component, it passes JSX attributes to this component as a single object called 'props'.
+                    Props are read-only, meaning they cannot be modified once packaged together. However, they can still be accessed within the child component. All React
+                    components MUST act like pure functions(they do not attempt to change their inputs) with respect to their props.
```

Comment:
Yea, this is where OOP and FP get a bit intermixed in JS as a Prototypical language.